client/data/domains/redirects/use-delete-domain-redirect-mutation.ts

```diff
@@ -0,0 +1,37 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { mapKeys, snakeCase } from 'lodash';
+import { useCallback } from 'react';
+import { DomainsApiError } from 'calypso/lib/domains/types';
+import wp from 'calypso/lib/wp';
+
+function toApi( data: Record< string, unknown > ) {
+	return mapKeys( data, ( v, k ) => snakeCase( k ) );
+}
+
+export default function useDeleteDomainRedirectMutation(
+	domainName: string,
+	queryOptions: {
+		onSuccess?: ( ...args: unknown[] ) => void;
+		onError?: ( error: DomainsApiError ) => void;
```

Comment:
I changed `...args: unknown[]` to `error: DomainsApiError` to fix the TS error. Would like to confirm will we be passing other args?

Reply:
We might want to form our own, I don't think we use a specific domains error on the backend this might not be the same thing at least conceptually.